tests/Unit/ApparentHorizons/Test_ComputeHorizonVolumeQuantities.cpp

```diff
@@ -0,0 +1,476 @@
+                                  (Phi.get(i, j + 1, c) + Phi.get(j, i + 1, c));
+        }
+      }
+    }
+    Pi.get(0, 0) = std::numeric_limits<double>::signaling_NaN();
+  }
+
+  if constexpr (tmpl::list_contains_v<
+                    SrcTags, Tags::deriv<GeneralizedHarmonic::Tags::Phi<
+                                             3, Frame::Inertial>,
+                                         tmpl::size_t<3>, Frame::Inertial>>) {
+    // Need to compute numerical deriv of Phi.
+    // The partial_derivatives function allows differentiating only a
+    // subset of a Variables, but in that case, the resulting Variables
+    // cannot point into the original Variables, and also the subset must
+    // be the tags that occur at the beginning of the Variables.  Because
+    // this is only a test, we just create new Variables and copy.
+
+    // vars to be differentiated
+    using phi_tag_list =
+        tmpl::list<::GeneralizedHarmonic::Tags::Phi<3, Frame::Inertial>>;
+    Variables<phi_tag_list> vars_before_differentiation(
+        mesh.number_of_grid_points());
+    get<::GeneralizedHarmonic::Tags::Phi<3, Frame::Inertial>>(
+        vars_before_differentiation) =
+        get<::GeneralizedHarmonic::Tags::Phi<3, Frame::Inertial>>(src_vars);
+
+    // differentiate
+    const auto vars_after_differentiation = partial_derivatives<phi_tag_list>(
+        vars_before_differentiation, mesh, inv_jacobian_logical_to_inertial);
+
+    // copy to src_vars.
+    get<Tags::deriv<GeneralizedHarmonic::Tags::Phi<3, Frame::Inertial>,
+                    tmpl::size_t<3>, Frame::Inertial>>(src_vars) =
+        get<Tags::deriv<GeneralizedHarmonic::Tags::Phi<3, Frame::Inertial>,
+                        tmpl::size_t<3>, Frame::Inertial>>(
+            vars_after_differentiation);
+  }
+
+  // Compute dest_vars
+  Variables<DestTags> dest_vars(mesh.number_of_grid_points());
+  if constexpr (IsTimeDependent::value) {
+    ah::ComputeHorizonVolumeQuantities::apply(
+        make_not_null(&dest_vars), src_vars, mesh, jacobian_target_to_inertial,
+        inv_jacobian_logical_to_target);
+  } else {
+    // time-independent.
+    ah::ComputeHorizonVolumeQuantities::apply(make_not_null(&dest_vars),
+                                              src_vars, mesh);
+  }
+
+  // Now make sure that dest vars are correct.
+
+  const auto expected_christoffel_second_kind = raise_or_lower_first_index(
+      gr::christoffel_first_kind(d_g), determinant_and_inverse(g).second);
+  const auto& christoffel_second_kind =
+      get<gr::Tags::SpatialChristoffelSecondKind<3, TargetFrame>>(dest_vars);
+  if constexpr (not std::is_same_v<TargetFrame, Frame::Inertial>) {
+    // Use a more forgiving local_approx because for the multi-frame case,
+    // christoffel_second_kind is computed using numerical derivatives,
+    // therefore this test should agree to truncation error and not roundoff.
+    Approx local_approx = Approx::custom().epsilon(1.e-5).scale(1.);
```

Comment:
Does this approx really need to be this loose? Can it be tighter with more grid points? I'm slightly concerned that 1e-5 could be large enough that it doesn't really check for all possible mistakes -- especially when the leading effect of the translation map will already be down around ~1e-2.

Reply:
Good catch. I don't know why I went as low as 1.e-5 (maybe it was an earlier version where I had fewer grid points or a larger domain). I was able to change to 1e-9 with the same number of grid points (8), and I can go to 1e-10 with 9 grid points.  So I changed it to 8 points and 1e-9.